infra/aws/govcloud.md

```diff
@@ -0,0 +1,199 @@
+Account" screen with the pulldown in the upper right of the AWS dashboard,
+and scroll down; there will be a "Sign up for AWS GovCloud" button. When
+you click that, you will need to accept an appendix to the AWS Customer
+Agreement; once you do so, you will get an email with instructions on how
+to log in to your GovCloud account.
+
+Once you create the account, you can create a temporary IAM user with full
+credentials you can use to bootstrap Terraform in the account as you would
+in commercial AWS.
+
+Once you have bootstrapped Terraform, you can create the GovCloud
+organization the same way you would in commercial; however, you cannot
+add `cloudtrail.amazonaws.com` as a service access principal in your
+organization definition due to the way Cloudtrail is automatically added
+to all GovCloud accounts.
+
+### Creating Additional GovCloud Accounts
+
+To add more accounts to your GovCloud organization, you *cannot* simply
+use the Terraform `aws_organizations_account` resource. Using credentials
+for the master account in your commercial AWS Organization, use the
+following AWS CLI command. This will create two accounts; a dummy account
+in the commercial AWS Organization and a new corresponding GovCloud
+account, both with the same account alias.
+
+```console
+$ aws organizations create-gov-cloud-account \
+  --email spacecats-infra+spacecats-govcloud-id@example.com \
+  --account-name spacecats-govcloud-id
+```
+
+(You can add `--iam-user-access-to-billing DENY` if you want, but this
+won't really matter once we lock down the dummy account later.)
+
+This will give you output that looks something like this:
+
+```json
+{
+  "CreateAccountStatus": {
+    "Id": "car-8d813980526511ea7184124ac536023e",
+    "AccountName": "spacecats-govcloud-id",
+    "State": "IN_PROGRESS",
+    "RequestedTimestamp": 1582040671.363
+  }
+}
+```
+
+You can then run the following command to get the account ID of your new
+GovCloud account:
+
+```console
+$ aws organizations describe-create-account-status \
+  --create-account-request-id car-8d813980526511ea7184124ac536023e
+```
+
+You'll get output that looks like this:
+
+```json
+{
+  "CreateAccountStatus": {
+    "Id": "car-8d813980526511ea7184124ac536023e",
+    "AccountName": "spacecats-govcloud-id",
+    "State": "SUCCEEDED",
+    "RequestedTimestamp": 1582040671.446,
+    "CompletedTimestamp": 1582040736.794,
+    "AccountId": "123456789000",
+    "GovCloudAccountId": "000987654321"
+  }
+}
+```
+
+In that output, `AccountId` is the account number for the dummy account,
+and the `GovCloudAccountId` is the account number for your new GovCloud
+account.
+
+Now that you have that, you can assume the `OrganizationAccountAccessRole`
+in the new account from an account with sufficient privileges in the
+*GovCloud* organization master account. *Note that this is the case even
+though at this point the GovCloud organization does not show up if you
+look at the organization membership in the console.*
```

Comment:
An example of assuming roles would be nice. Are you doing it from the CLI or via the AWS config file?

Reply:
I usually just add a profile for it in `.aws/config` -- that tends to be easier, since `aws-vault` will pull it for all our `aws` and `terraform` commands.